cinder/volume/api.py

```diff
@@ -1669,7 +1669,22 @@ def migrate_volume(self, context, volume, host, cluster_name, force_copy,
         else:
             expected['host'] = db.Not(host)
 
-        filters = [~db.volume_has_snapshots_filter()]
+        # SAP: If the backend's snapshots are clones, then we don't
+        # have to filter out volumes that have snapshots.  We can find out
+        # if a backend has clones for snapshots by fetching the backend
+        # capabilities and looking for snapshot_type=='clone'
+        caps = self.volume_rpcapi.get_capabilities(
+            context, volume.service_topic_queue, False)
```

Comment:
Please add a keyword to naked boolean parameters, it's pretty opaque otherwise what they mean:
```suggestion
            context, volume.service_topic_queue, discover=False)
```

Reply:
I was simply following the conventions for everywhere else in Cinder that calls get_capabilities.  Every other instance of a call doesn't used named params.

https://github.com/sapcc/cinder/blob/stable/wallaby-m3/cinder/cmd/manage.py#L109

https://github.com/sapcc/cinder/blob/stable/wallaby-m3/cinder/volume/manager.py#L2382-L2384

Anyway, I will change it.

Reply:
I have updated the filter to conditionally include the snaps text if and only if there are snapshot filters in place.